### ISSUE TITLE
[Bug]: Crash when trying to redirect to Netdata Cloud in dashboard

### ISSUE THREAD

**Issue by hakula139** [netdata/netdata#15259]
### Bug description

When I clicked "Sign in" in the Netdata dashboard, it took me to the login page, and I logged in. Then in the dashboard I found it's still requiring me to log in. When I click "Sign in" again, it just flashes the login page and immediately redirects back.

<img width="1919" alt="image" src="https://github.com/netdata/netdata/assets/9496568/c26bcac5-28e4-44b6-96ab-4435be6dd3d0">

Note that in <https://app.netdata.cloud> I can find myself logged in, and can find the active node.

<img width="1919" alt="image" src="https://github.com/netdata/netdata/assets/9496568/6ba24649-e744-4e2d-88d0-a19fc172002e">

Now I click "Take me to Netdata Cloud" in the dashboard, it redirects to the login page and redirects back, then I get a 502 error. When checking the service status on my server, I've found netdata.service has just crashed.

```text
● netdata.service - Real time performance monitoring
     Loaded: loaded (/lib/systemd/system/netdata.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: signal) since Tue 2023-06-27 15:48:09 CST; 4s ago
    Process: 135632 ExecStart=/usr/sbin/netdata -D $EXTRA_OPTS (code=killed, signal=SEGV)
   Main PID: 135632 (code=killed, signal=SEGV)
        CPU: 4.781s

Jun 27 15:48:09 hakula systemd[1]: netdata.service: Consumed 4.781s CPU time.
```

### Expected behavior

1. In the dashboard, it should be in logged-in status.
2. Then I can redirect to the Netdata Cloud page without making the netdata.service crash.

### Steps to reproduce

1. Log in to Netdata Cloud.
2. Click "Take me to Netdata Cloud" button in the dashboard.
3. netdata.service crashed.

### Installation method

manual setup of official DEB/RPM packages

### System info

```shell
Linux hakula 5.10.0-20-amd64 #1 SMP Debian 5.10.158-2 (2022-12-13) x86_64 GNU/Linux
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="11"
/etc/os-release:VERSION="11 (bullseye)"
/etc/os-release:VERSION_CODENAME=bullseye
/etc/os-release:ID=debian
```

### Netdata build info

```shell
Version: netdata v1.40.0-45-nightly
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-option-checking' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' '--disable-dependency-tracking' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -ffile-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -ffile-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Install type: binpkg-deb
    Binary architecture: x86_64
    Packaging distro:  
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK:                       YES
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
    HTTPD (h2o):                YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    debugfs:                 YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
Debug/Developer Features:
    Trace Allocations:       NO
```

### Additional info

It seems only related to recent nightly versions. I did not encounter this issue before perhaps `v1.40.0-39-nightly` (not sure).

```bash
$ journalctl -exfu netdata
```

```text
Jun 27 15:48:08 hakula apps.plugin[135830]: Received error on stdin.
Jun 27 15:48:08 hakula systemd[1]: netdata.service: Main process exited, code=killed, status=11/SEGV
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit netdata.service has exited.
░░ 
░░ The process' exit code is 'killed' and its exit status is 11.
Jun 27 15:48:08 hakula ebpf.plugin[135822]: thread with task id 135854 finished
Jun 27 15:48:08 hakula ebpf.plugin[135822]: thread with task id 135857 finished
Jun 27 15:48:08 hakula ebpf.plugin[135822]: thread with task id 135855 finished
Jun 27 15:48:09 hakula ebpf.plugin[135822]: thread with task id 135856 finished
Jun 27 15:48:09 hakula ebpf.plugin[135822]: thread with task id 135860 finished
Jun 27 15:48:09 hakula ebpf.plugin[135822]: thread with task id 135862 finished
Jun 27 15:48:09 hakula ebpf.plugin[135822]: thread with task id 135858 finished
Jun 27 15:48:09 hakula ebpf.plugin[135822]: thread with task id 135859 finished
Jun 27 15:48:09 hakula systemd[1]: netdata.service: Failed with result 'signal'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit netdata.service has entered the 'failed' state with result 'signal'.
Jun 27 15:48:09 hakula systemd[1]: netdata.service: Consumed 4.781s CPU time.
░░ Subject: Resources consumed by unit runtime
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit netdata.service completed and consumed the indicated resources.
Jun 27 15:48:39 hakula systemd[1]: netdata.service: Scheduled restart job, restart counter is at 11.
░░ Subject: Automatic restarting of a unit has been scheduled
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ Automatic restarting of the unit netdata.service has been scheduled, as the result for
░░ the configured Restart= setting for the unit.
```

There's no log between `2023-06-27 15:47:34` and `2023-06-27 15:48:39` in `netdata/error.log`.

**Comment by hakula139** [comment]
It seems that clicking "Check Now" can also crash the service.

**Comment by MrZammler** [comment]
Hi @hakula139.

I can confirm that there is something weird when doing the steps mentioned above. In my case however, the agent doesn't crash, but indeed goes back to the local dashboard instead of actually logging in to the cloud.

**Comment by MrZammler** [comment]
Does `coredumpctl` provide any info on the crash?

**Comment by hakula139** [comment]
> Does `coredumpctl` provide any info on the crash?

Well I'm not sure if I'm using it correctly, but it says there's no coredumps found.

**Comment by hakula139** [comment]
> Does `coredumpctl` provide any info on the crash?

Here's the coredump file:

[coredump.tar.xz.tar.gz](https://github.com/netdata/netdata/files/11881713/coredump.tar.xz.tar.gz)

**Comment by hakula139** [comment]
Fixed in version `v1.41.0-2-nightly`.
